[CI] Add secrets input for publish-to-bcr workflow

WojciechMazur opened the pull request:
Fixes https://github.com/EngFlow/gazelle_cc/pull/53#issuecomment-2892221463 found in test release https://github.com/EngFlow/gazelle_cc/actions/runs/15122752957

jayconrod reviewed (commented):
Same question here though: can we call it `ENGFLOW_AUTOMATION_PUBLIC_PAT` instead? It looks like we pass in the value of the token on L40, so I wouldn't expect it to need a particular name.

WojciechMazur commented:
I've renamed the secret to `ENGFLOW_AUTOMATION_PUBLIC_PAT` since it's value can be taken from workflow_call inputs (when referenced from release.yaml) or from repository/organization secrets directlly when triggered manually with workflow_dispatch, all of the usages should be named the same. 
With these changes the CI can successfully execute until the point of creating a PR to BCR, which fails due to lack of valid `publish_token` 

I've spotted one more problem - the action that we use expects `.bcr/presubmit.yml` instead of `.bcr/presubmit.yaml` - I've renamed this file to fix the issue
